diff -uprN qt-creator-opensource-src-4.5.0-orig/src/plugins/projectexplorer/projectexplorer.cpp qt-creator-opensource-src-4.5.0/src/plugins/projectexplorer/projectexplorer.cpp
--- qt-creator-opensource-src-4.5.0-orig/src/plugins/projectexplorer/projectexplorer.cpp	2017-12-01 11:22:06.000000000 +0100
+++ qt-creator-opensource-src-4.5.0/src/plugins/projectexplorer/projectexplorer.cpp	2017-12-07 18:44:12.333492300 +0100
@@ -2050,7 +2050,7 @@ void ProjectExplorerPluginPrivate::execu
         return;
     }
 
-    emit m_instance->aboutToExecuteProject(runConfiguration->target()->project(), runMode);
+    emit m_instance->aboutToExecuteRunControl(runControl, runMode);
 
     startRunControl(runControl);
 }
diff -uprN qt-creator-opensource-src-4.5.0-orig/src/plugins/projectexplorer/projectexplorer.h qt-creator-opensource-src-4.5.0/src/plugins/projectexplorer/projectexplorer.h
--- qt-creator-opensource-src-4.5.0-orig/src/plugins/projectexplorer/projectexplorer.h	2017-12-01 11:22:06.000000000 +0100
+++ qt-creator-opensource-src-4.5.0/src/plugins/projectexplorer/projectexplorer.h	2017-12-07 18:44:12.333492300 +0100
@@ -170,7 +170,7 @@ signals:
     // or the file list of a specific project has changed.
     void fileListChanged();
 
-    void aboutToExecuteProject(ProjectExplorer::Project *project, Core::Id runMode);
+    void aboutToExecuteRunControl(ProjectExplorer::RunControl *runControl, Core::Id runMode);
     void recentProjectsChanged();
 
     void settingsChanged();
