diff -uprN qt-creator-opensource-src-4.4.0-orig/src/plugins/projectexplorer/projectexplorer.cpp qt-creator-opensource-src-4.4.0/src/plugins/projectexplorer/projectexplorer.cpp
--- qt-creator-opensource-src-4.4.0-orig/src/plugins/projectexplorer/projectexplorer.cpp	2017-09-10 13:54:45.257643481 +0200
+++ qt-creator-opensource-src-4.4.0/src/plugins/projectexplorer/projectexplorer.cpp	2017-10-03 21:12:15.199877692 +0200
@@ -2012,7 +2012,7 @@ void ProjectExplorerPluginPrivate::execu
     auto runControl = new RunControl(runConfiguration, runMode);
     (void) producer(runControl);
 
-    emit m_instance->aboutToExecuteProject(runConfiguration->target()->project(), runMode);
+    emit m_instance->aboutToExecuteRunControl(runControl, runMode);
 
     startRunControl(runControl);
 }
diff -uprN qt-creator-opensource-src-4.4.0-orig/src/plugins/projectexplorer/projectexplorer.h qt-creator-opensource-src-4.4.0/src/plugins/projectexplorer/projectexplorer.h
--- qt-creator-opensource-src-4.4.0-orig/src/plugins/projectexplorer/projectexplorer.h	2017-09-10 13:54:45.257643481 +0200
+++ qt-creator-opensource-src-4.4.0/src/plugins/projectexplorer/projectexplorer.h	2017-10-03 21:13:18.091338932 +0200
@@ -166,7 +166,7 @@ signals:
     // or the file list of a specific project has changed.
     void fileListChanged();
 
-    void aboutToExecuteProject(ProjectExplorer::Project *project, Core::Id runMode);
+    void aboutToExecuteRunControl(ProjectExplorer::RunControl *runControl, Core::Id runMode);
     void recentProjectsChanged();
 
     void settingsChanged();
